arch/arm/core/aarch32/mpu/arm_mpu.c

```diff
@@ -55,7 +55,13 @@ static inline uint8_t get_num_regions(void)
 	return NUM_MPU_REGIONS;
 #else
 
-	uint32_t type = MPU->TYPE;
+	uint32_t type;
+
+#if defined(CONFIG_CPU_CORTEX_R)
+	__asm__ volatile("mrc p15, 0, %0, c0, c0, 4" : "=r" (type) ::);
```

Comment:
style: can you encapsulate these snippets of ASM into their own (maybe inline) functions? MISRA-C I believe requires this.

I think it would be good to have some accessor methods for these `mrc` operations

Reply:
It looks like you did this in arm_mpu_internal.h so maybe, more like that